README.md

```diff
@@ -104,6 +104,23 @@ extension scenarios:
 * Building a custom sampler for
   [traces](./docs/trace/extending-the-sdk/README.md#sampler).
 
+## Experimental Features
+
```

Comment:
Experimental features are enabled in following ways. 

**Features that require use of public APIs**
Public APIs that are available on experimental basis will have a clear API documentation specifying that.
(Example would be great here)


**Features that provide new functionality or changes the internal behavior without involvement of public APIs**
Such features can be used by opting in via environment variables (e.g., `OTEL_DOTNET_EXPERIMENTAL_SOME_FEATURE`)

**NOTE**: Both the ways will only be available in pre-release versions of the libraries. Stable versions will not include the ability to opt-in into experimental features. Users opted into use experimental features may face build time errors or runtime behavior change if they upgrade to stable version.

Reply:
After reviewing this again, I feel that we should have this clear demarcation of which kind of experimental features get offered in what ways. I agree with the above comment made by @vishweshbankwar that:
- any experimental feature that involves public API changes should be offered using the compiler attribute approach
- any purely internal behavior change should be offered using the `OTEL_DOTNET_EXPERIMENTAL_SOME_FEATURE` env vars (I would be opposed to offering them using the compiler attribute feature; I can talk more about my reasons if needed)

I would like to update the note to allow for purely internal behavior changes to be allowed to be shipped as part of the stable release. These are opt-in experimental features. There's no good reason to exclude them from stable releases.

> NOTE: ~~Both the ways will only be available in pre-release versions of the libraries. Stable versions will not include the ability to opt-in into experimental features. ~~ Users who opted in to use experimental features may face build time errors or runtime behavior change if they upgrade to stable version. Internal behavioral changes offered as part of feature flags maybe included in the stable versions as well.

@alanwest To your [concern](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4762#discussion_r1290838715) of the attributes' discussion blocking progress, I'd be okay with commenting out the code in the OTLP Exporter for exporting the three attributes for now and address it after this PR is merged.